build-system/common/exec.js

```diff
@@ -20,6 +20,8 @@
  */
 
 const childProcess = require('child_process');
+const log = require('fancy-log');
+const {yellow} = require('ansi-colors');
```

Comment:
As discussed offline, these two package dependencies will indirectly break `check-package-manager.js`, which is expected to be able to run before any packages are installed.

The real problem is not that you are using packages here (perfectly legitimate), but that `check-package-manager.js` depends on `exec.js` for `getStdout` and `getStderr`.

On looking at both files, I think we should move `getOutput`, `getStdout`, `getStderr`, and `spawnProcess` into a separate file (maybe call it `output.js`?) with a comment at the top that it should not depend on any packages installed to `node_modules/`. This will free up `exec.js` to use whatever packages it wishes to. WDYT?

Reply:
That SGTM, since those helpers don't have any dependencies themselves. I'm inclined to call it something more like `process.js` though.

Reply:
Done :+1: